src/Profile/EditAvatarScreen.js

```diff
@@ -77,83 +95,77 @@ export default function EditAvatarScreen({ route, navigation }) {
   const [bokchoy, setBokchoy] = useState(avatarNum === 6);
 
   useEffect(() => {
-    const useremail = 'helen@gmail.com';
-    base('Users').select({
-      filterByFormula: `({email}='${useremail}')`,
-    }).firstPage().then((record) => {
-      switch (record[0].fields.avatarNum) {
-        case 1: setAvatar(require('../assets/imgs/pipa.png'));
-          setAvatarNum(0);
-          setPipa(true);
-          break;
-        case 2: setAvatar(require('../assets/imgs/eggplant.png'));
-          setAvatarNum(1);
-          setEggplant(true);
-          break;
-        case 3: setAvatar(require('../assets/imgs/mango.png'));
-          setAvatarNum(2);
-          setMango(true);
-          break;
-        case 4: setAvatar(require('../assets/imgs/dragonfruit.png'));
-          setAvatarNum(3);
-          setDragonfruit(true);
-          break;
-        case 5: setAvatar(require('../assets/imgs/lychee.png'));
-          setAvatarNum(4);
-          setLychee(true);
-          break;
-        case 6: setAvatar(require('../assets/imgs/bokchoy.png'));
-          setAvatarNum(5);
-          setBokchoy(true);
-          break;
-        default: setAvatar(require('../assets/imgs/placeholder.png'));
-      }
-    });
+    switch (currentUser.avatarNum) {
+      case 1: setAvatar(pipaIMG);
+        setAvatarNum(0);
+        setPipa(true);
+        break;
+      case 2: setAvatar(eggplantIMG);
+        setAvatarNum(1);
+        setEggplant(true);
+        break;
+      case 3: setAvatar(mangoIMG);
+        setAvatarNum(2);
+        setMango(true);
+        break;
+      case 4: setAvatar(dragonfruitIMG);
+        setAvatarNum(3);
+        setDragonfruit(true);
+        break;
+      case 5: setAvatar(lycheeIMG);
+        setAvatarNum(4);
+        setLychee(true);
+        break;
+      case 6: setAvatar(bokchoyIMG);
+        setAvatarNum(5);
+        setBokchoy(true);
+        break;
+      default: setAvatar(placeholderIMG);
+    }
   }, []);
 
-  const sendUpdate = async () => {
-    const user = 'recIpBFqr2EXNbS7d';
-    await base('Users').update([
-      {
-        id: user,
-        fields: {
-          avatarNum,
-        },
-      },
-    ]);
-
-    route.params.setRefresh(route.params.refresh + 1);
-    navigation.goBack();
+  const cancelAvatar = () => {
+    navigation.goBack(null);
   };
 
   const saveAvatar = async () => {
-    const user = 'recIpBFqr2EXNbS7d';
-    await base('Users').find(user, (err, record) => {
-      if (err) {
-        return;
-      }
-
-      sendUpdate();
-    });
+    try {
+      await base('Users').update([
+        {
+          id: currentUser.id,
+          fields: {
+            avatarNum,
+          },
+        },
+      ]);
+      const updatedUser = { ...currentUser, avatarNum };
+      serviceUpdateUser(updatedUser);
+      setRefresh(refresh + 1);
```

Comment:
replace refresh/setRefresh variables with avatar/setAvatart instead, simplifies the whole process since you won't need to duplicate existing functions